libraries/chain/fork_database.cpp

```diff
@@ -804,6 +804,16 @@ namespace eosio::chain {
       in_use = in_use_t::savanna;
    }
 
+   // only called from the main thread
+   void fork_database::switch_to_legacy() {
+      in_use = in_use_t::legacy;
+   }
+
+   // only called from the main thread
+   void fork_database::switch_to_both() {
+      in_use = in_use_t::both;
+   }
```

Comment:
What about a single function: `fork_database::switch_to(in_use_t v);`?

I'd add this to controller_impl: `using in_use_t = fork_database::in_use_t;`

Reply:
Originally didn't need all three. At this point, might as well.